tests/topotests/bgp-evpn-vxlan_topo1/test_bgp_evpn_vxlan.py

```diff
@@ -0,0 +1,275 @@
+    # This function initiates the topology build with Topogen...
+    tgen = Topogen(TemplateTopo, mod.__name__)
+    # ... and here it calls Mininet initialization functions.
+    tgen.start_topology()
+
+    pe1 = tgen.gears['PE1']
+    pe2 = tgen.gears['PE2']
+    p1 = tgen.gears['P1']
+    
+    # set up PE bridges with the EVPN member interfaces facing the CE hosts
+    pe1.run('ip link add name br101 type bridge stp_state 0')
+    pe1.run('ip link set dev br101 up')
+    pe1.run('ip link add vxlan101 type vxlan id 101 dstport 4789 local 1.1.1.1 nolearning')
+    pe1.run('ip link set dev vxlan101 master br101')
+    pe1.run('ip link set up dev vxlan101')
+    pe1.run('ip link set dev PE1-eth0 master br101')
+
+    pe2.run('ip link add name br101 type bridge stp_state 0')
+    pe2.run('ip link set dev br101 up')
+    pe2.run('ip link add vxlan101 type vxlan id 101 dstport 4789 local 3.3.3.3 nolearning')
+    pe2.run('ip link set dev vxlan101 master br101')
+    pe2.run('ip link set up dev vxlan101')
+    pe2.run('ip link set dev PE2-eth1 master br101')
+    p1.run('sysctl -w net.ipv4.ip_forward=1')
+
+    # This is a sample of configuration loading.
+    router_list = tgen.routers()
+
+    # For all registred routers, load the zebra configuration file
+    for rname, router in router_list.iteritems():
+        router.load_config(
+            TopoRouter.RD_ZEBRA,
+            # Uncomment next line to load configuration from ./router/zebra.conf
+            os.path.join(CWD, '{}/zebra.conf'.format(rname))
+        )
+        router.load_config(
+            TopoRouter.RD_OSPF,
+            # Uncomment next line to load configuration from ./router/zebra.conf
+            os.path.join(CWD, '{}/ospfd.conf'.format(rname))
+        )
+        router.load_config(
+            TopoRouter.RD_BGP,
+            # Uncomment next line to load configuration from ./router/zebra.conf
+            os.path.join(CWD, '{}/bgpd.conf'.format(rname))
+        )
+
+    # After loading the configurations, this function loads configured daemons.
+    tgen.start_router()
+
+def teardown_module(mod):
+    "Teardown the pytest environment"
+    tgen = get_topogen()
+
+    # This function tears down the whole topology.
+    tgen.stop_topology()
+
+def test_pe1_converge_evpn():
+    "Wait for protocol convergence"
+
+    tgen = get_topogen()
+    # Don't run this test if we have any failure.
+    if tgen.routers_have_failure():
+        pytest.skip(tgen.errors)
+
+    pe1 = tgen.gears['PE1']
+    json_file = '{}/{}/evpn.vni.json'.format(CWD, pe1.name)
+    expected = json.loads(open(json_file).read())
+
+    test_func = partial(topotest.router_json_cmp,
+                        pe1, 'show evpn vni 101 json', expected)
+    _, result = topotest.run_and_expect(test_func, None, count=125, wait=1)
+    assertmsg = '"{}" JSON output mismatches'.format(pe1.name)
+    assert result is None, assertmsg
+    #tgen.mininet_cli()
+
+def test_pe2_converge_evpn():
```

Comment:
I see this method is identical to test_pe1_converge_evpn. can you squash to a single one and add tgen.gears['..'] as a variable for this method?

Reply:
I can squash both into a single test but I'd kind of like it to be 2.